packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx

```diff
@@ -179,6 +190,29 @@ function useHighlightState({
   return state;
 }
 
+/**
+ * Hook to handle PDF render error
+ */
+
+export function useIsPfdError(
+  isPdfRenderError: boolean,
+  callbackIsPdfError?: (state: boolean) => any
+) {
+  const [currentIsPdfError, setCurrentIsPdfError] = useState(isPdfRenderError);
+  console.log('useIsPfdError', currentIsPdfError, callbackIsPdfError);
+
+  // Process error message
+  const _previouserrMsgFromPdf = useRef(isPdfRenderError);
+  useEffect(() => {
+    console.log('_previouserrMsgFromPdf', _previouserrMsgFromPdf);
+    console.log('currentErrMsgFromPdf  ', currentIsPdfError);
+    console.log('localErrMsgFromPdf    ', isPdfRenderError);
+    callbackIsPdfError?.(currentIsPdfError);
```

Comment:
What is this line doing?

Reply:
Calling `callbackIsPdfError` is updating the state and passing the state change from PDFViewer (where error happened) to DocumentPreview (logic changes display from pdf to html/text). PdfViewerWithHighlight is an intermediate component.